.github/workflows/nix_build_linux.yml

```diff
@@ -1,431 +1,318 @@
+        run: |
+          chmod +x ./examples/copy_artifacts.sh
+          ./examples/copy_artifacts.sh build/examples/cpp examples_output
+          ls -al examples_output/
+
+      - name: Set aritfacts' names
+        id: artifact-names
+        run: |
+          echo "transpiler=transpiler-output" >> $GITHUB_OUTPUT
+          echo "examples=circuits-and-assignments" >> $GITHUB_OUTPUT
+
+      - name: Upload examples' circuits and assignments artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.artifact-names.outputs.examples }}
+          path: |
+            examples_output
+
+      - name: Compile tests as cpp code
+        run: |
+          nix develop -c ninja -C build -k 0 all_tests_compile_as_cpp_code
+
+      - name: Compile tests as circuits
+        run: |
+          nix develop -c ninja -C build -k 0 all_tests_compile_as_circuits
+
+      - name: Run tests as cpp code (expected res calculation)
+        run: |
+          nix develop -c ninja -C build -k 0 all_tests_run_expected_res_calculation
+
+      - name: Get transpiler output for integration testing
+        run: |
+          targets_str=$(echo "${{ env.INTEGRATION_TESTING_TARGETS }}" | awk 'NF {printf " %s_evm_verifier ", $0}' | sed 's/[[:space:]]*$//')
+          echo "targets: ${targets_str}"
+          nix develop -c ninja -C build -k 0  ${targets_str}
+          ls -al build/examples/cpp/
+
+      - name: Upload examples' circuits and assignments artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.artifact-names.outputs.transpiler }}
+          path: |
+            build/examples/cpp/transpiler_output_*
+
+      - name: Run tests as circuits (real res calculation)
+        run: |
+          nix develop -c ninja -C build -k 0 all_tests_assign_circuits
+
+      - name: Compare expected and real test results
+        run: |
+          chmod +x ./tests/run_test_results_comparison.sh
+          bash ./tests/run_test_results_comparison.sh
+
+      - name: Run tests on faulty inputs
+        run: |
+          chmod +x ./tests/run_tests_faulty_input.sh
+          bash ./tests/run_tests_faulty_input.sh pallas
+
+      - name: Run assigner unit tests
+        run: |
+          nix develop -c ninja -C build -k 0 check-crypto3-assigner
+
+      - name: Build proof for the circuit of the C++ examples
+        run: |
+          nix develop -c ninja -C build -k 0 prove_cpp_examples
+
+      - name: Build recursive gen
+        run: |
+          nix develop -c ninja -C build -k 0 recursive_gen
+
+      - name: Build and run transpiler test
+        run: |
+          nix develop -c ninja -C build -k 0 compile_and_run_transpiler_tests
+
+      - name: Run recursive verifier tests
+        run: |
+          nix develop -c ninja -C build -k 0 recursion
+
+      - name: Build IR of the C++ examples
+        run: |
+          nix develop -c ninja -C build -k 0 compile_cpp_examples

```

Comment:
Step should be placed between "Build zkllvm" and "Build circuits(.crct) of the C++ examples", as it was earlier

Reply:
done